lib/livebook/runtime/evaluator.ex

```diff
@@ -688,20 +692,100 @@ defmodule Livebook.Runtime.Evaluator do
     {result, code_markers}
   end
 
-  defp eval(:erlang, code, binding, env) do
+  # Erlang code is either statements as currently supported, or modules.
+  # In case we want to support modules - it makes sense to allow users to use
+  # includes, defines and thus we use the epp-module first - try to find out
+  #
+  # if in the tokens from erl_scan we find at least 1 module-token we assume
+  # that the user is defining a module, if not the previous code is called.
+
+  defp eval(:erlang, code, binding, env, tmp_dir) do
+    case :erl_scan.string(String.to_charlist(code), {1, 1}, [:text]) do
+      {:ok, tokens, _} ->
+        case :lists.keyfind(:module, 3, tokens) do
```

Comment:
I think this is checking for AST but you need to check tokens. You will have to revert to some code you had earlier. :)

Reply:
I tried to isolate the code a bit more since if we have no filesystem the normal Erlang-statements would still work.

Reply:
Oh, I see what you mean. Can we have false positives? Could this match for example on an atom called `:module`? Perhaps it is best to match on the tokens `-` and `module`? But even that can be a mismatch on `Foo-module` (sure, it is invalid Erlang). Maybe we should just enforce that it must start with `-module`? So we check only the first tokens?